Add /mail clear <player> and /mail clearall commands

Division7 opened the pull request:
### Information

<!--
    Replace #nnnn with the number of the original issue. If this PR implements
    features from multiple issues, you should repeat the phrase "closes #nnnn"
    for each issue. 
-->

This PR closes #2057. 

### Details

**Proposed feature:**  
<!-- Type a description of your proposed feature below this line. -->
/mail clear [player] and /mail clearall. Functions on online and offline players, and clears their mailboxes. Adds permissions essentials.mail.clear.others and essentials.mail.clearall.

Note; please look closely at my code. This is one of my first times working on a GitHub project, so I want to be sure I'm not breaking anything and that I'm doing it right.

**Environments tested:**    

<!-- Type the OS you have used below. -->
OS: Windows 11

<!-- Type the JDK version (from java -version) you have used below. -->
Java version:  Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39)

<!--
    Put an "x" inside the boxes for the server software you have tested this 
    bug fix on. If this feature does not apply to a server, strike through the server software using ~~strikethrough~~. If you have tested on other
    environments, add a new line with relevant details.
-->
- [ ] Most recent Paper version (1.XX.Y, git-Paper-BUILD)
- [x] CraftBukkit/Spigot/Paper 1.18.2
- [ ] CraftBukkit 1.8.8


**Demonstration:**  
<!--
    Below this block, include screenshots/log snippets from before and after as
    necessary. If you have created or used a test case plugin, please link to a
    download of the plugin, source code and exact version used where possible.
-->

Before /mail clearall:
![2022-04-01_19 05 58](https://user-images.githubusercontent.com/26347897/161361563-e81d0f34-3520-4a94-9bac-991b550d3dc2.png)

After /mail clearall:
![2022-04-01_19 06 15](https://user-images.githubusercontent.com/26347897/161361584-e9189c14-4af2-4a4d-88ea-4fd633b2ab23.png)

Before /mail clear [player]:
![2022-04-01_19 06 34](https://user-images.githubusercontent.com/26347897/161361592-79fb4eb2-852b-4acb-9de8-d4ede1bcc9b9.png)

After /mail clear [player]:
![2022-04-01_19 06 47](https://user-images.githubusercontent.com/26347897/161361604-2eeb9559-0769-40d4-b3bc-f5659b27cb95.png)

Commit:
Added commands clearall and clear [player] to Mail command. Added their messages to messages.properties.

Commit:
bugfixed clearall and clear, updated messages_en.properties

Division7 committed:
Merge branch 'EssentialsX:2.x' into 2.x

Division7 committed:
Merge branch '2.x' into 2.x

Division7 commented:
I'm sorry, I'm still kind of new to GitHub. Is there something I need to do to get this approved?

mdcfe commented:
> I'm sorry, I'm still kind of new to GitHub. Is there something I need to do to get this approved?

We haven't had the time to review this PR yet. Once the 1.19 update is finished, we'll start reviewing PRs that add new features.

Division7 committed:
Merge branch '2.x' into 2.x

Division7 committed:
Merge branch '2.x' into 2.x

mdcfe commented on Essentials/src/main/java/com/earth2me/essentials/commands/Commandmail.java:
```diff
@@ -194,15 +194,53 @@ public void run(final Server server, final User user, final String commandLabel,
+            }
+            u.setMailList(null);
+            sender.sendMessage(tl("mailClearedPlayer", u.getDisplayName()));
+            return;
```
Move this to a separate method instead of copying it here.

mdcfe commented on Essentials/src/main/java/com/earth2me/essentials/commands/Commandmail.java:
```diff
@@ -286,9 +324,12 @@ protected List<String> getTabCompleteOptions(final Server server, final User use
             return options;
         } else if (args.length == 2) {
-            if ((args[0].equalsIgnoreCase("send") && user.isAuthorized("essentials.mail.send")) || (args[0].equalsIgnoreCase("sendtemp") && user.isAuthorized("essentials.mail.sendtemp"))) {
+            if ((args[0].equalsIgnoreCase("send") && user.isAuthorized("essentials.mail.send")) || (args[0].equalsIgnoreCase("sendtemp") && user.isAuthorized("essentials.mail.sendtemp")) || (args[0].equalsIgnoreCase("clear"))) {
```
There needs to be a permission check for `essentials.mail.clear.others` here

mdcfe commented on Essentials/src/main/resources/messages_en.properties:
```diff
@@ -675,20 +675,25 @@ loomCommandUsage=/<command>
-mailCommandUsage5Description=Sends the specified player the given message which will expire in the specified time
-mailCommandUsage6=/<command> sendtempall <expire time> <message>
-mailCommandUsage6Description=Sends all players the given message which will expire in the specified time
+mailCommandUsage3=/<command> clear [player]
+mailCommandUsage3Description=Clears the mail of the specified player.
+mailCommandUsage4=/<command> send <player> <message>
+mailCommandUsage4Description=Sends the specified player the given message
```
Don't edit `messages_en.properties` - it will get overridden by Crowdin.

mdcfe commented on Essentials/src/main/resources/messages.properties:
```diff
@@ -675,21 +675,27 @@ loomCommandDescription=Opens up a loom.
-mailCommandUsage4Description=Sends all players the given message
-mailCommandUsage5=/<command> sendtemp <player> <expire time> <message>
-mailCommandUsage5Description=Sends the specified player the given message which will expire in the specified time
-mailCommandUsage6=/<command> sendtempall <expire time> <message>
-mailCommandUsage6Description=Sends all players the given message which will expire in the specified time
+mailCommandUsage3=/<command> clear [player]
+mailCommandUsage3Description=Clears the mail of the specified player.
+mailCommandUsage4=/<command> send <player> <message>
+mailCommandUsage4Description=Sends the specified player the given message
+mailCommandUsage5=/<command> sendall <message>
+mailCommandUsage5Description=Sends all players the given message
+mailCommandUsage6=/<command> sendtemp <player> <expire time> <message>
+mailCommandUsage6Description=Sends the specified player the given message which will expire in the specified time
+mailCommandUsage7=/<command> sendtempall <expire time> <message>
+mailCommandUsage7Description=Sends all players the given message which will expire in the specified time
+mailCommandUsage8=/<command> clearall
+mailCommandUsage8Description=Clears the mail of all players
```
Right now this will delete existing translations in other languages. `clear [player]` should be added at the bottom, between `sendtempall` and `clearall`.

Division7 committed:
Merge branch 'EssentialsX:2.x' into 2.x

Division7 committed:
added requested changes.
new clearMail function returns user used on to make it easier to send the return message back to the correct source.

Division7 commented:
Alright, I believe I did what you asked. To make it easier to display the user the clear command was used on, I set the clearMail() command to return that User object so I can include their name in the message.

Division7 committed:
Merge branch '2.x' into 2.x

Division7 committed:
Merge branch '2.x' into 2.x

Division7 committed:
Merge branch '2.x' into 2.x

Division7 committed:
Merge branch '2.x' into 2.x

Division7 committed:
Merge branch '2.x' into 2.x

Division7 committed:
Merge branch '2.x' into 2.x

Division7 committed:
Merge branch '2.x' into 2.x

Division7 commented:
Thought I'd give this a poke. Not sure if I have to do anything from here.

Division7 committed:
Merge branch '2.x' into 2.x

JRoy committed:
Merge remote-tracking branch 'upstream/2.x' into mail-shit

JRoy committed:
Some changes

JRoy committed:
Discard changes to Essentials/src/main/resources/messages_en.properties

JRoy committed:
Merge branch '2.x' of https://github.com/Division7/Essentials into mail-shit

JRoy committed:
Some changes 2

JRoy committed:
MMMM
